crates/pop-cli/src/commands/test/contract.rs

```diff
@@ -22,13 +29,54 @@ pub(crate) struct TestContractCommand {
 
 impl TestContractCommand {
 	/// Executes the command.
-	pub(crate) fn execute(self) -> anyhow::Result<&'static str> {
+	pub(crate) async fn execute(mut self) -> anyhow::Result<&'static str> {
 		clear_screen()?;
 		if self.features.is_some() && self.features.clone().unwrap().contains("e2e-tests") {
+			warning("--features e2e-tests is deprecated. Use --e2e instead.")?;
+			self.e2e = true;
+			#[cfg(not(test))]
+			sleep(Duration::from_secs(3));
+		}
+
+		if self.e2e {
 			intro(format!(
 				"{}: Starting end-to-end tests",
 				style(" Pop CLI ").black().on_magenta()
 			))?;
+
+			// if the contracts node binary does not exist, prompt the user to download it
+			let maybe_contract_node_path = does_contracts_node_exist(crate::cache()?);
+			if maybe_contract_node_path == None {
+				warning("The substrate-contracts-node binary is not found. This is needed to run end-to-end tests.")?;
```

Comment:
Note that none of this triggered for me as I had s-c-n previously installed via cargo. Doing so resulted in a poor experience, so perhaps we only check the cache for the binary for a better experience? It downloads in seconds so think it better to only support pop-managed binaries.

Reply:
We had a discussion about this. We will leave in place for now as it can be nice to use the standalone binary if it exists.

However, there should be some improvements in a follow-up PR, including checking the version and downloading to the pop cache if out-of-date. cc @AlexD10S.

Reply:
With the changes in `does_contracts_node_exist` this is outdated now right? Test it with s-c-n installed via cargo and UX looks good.

Reply:
For @evilrobot-01, it did something weird where it starting building the binary again. For me, I had no problems with the standalone s c n binary from cargo.

Reply:
After working on the versioning PR: https://github.com/r0gue-io/pop-cli/pull/262 I am in favor of the first comment to only support pop-managed binaries. Supporting other versions of `substrate-contracts-node` makes versioning very hacky.

Reply:
Will be tackled in another PR